### PR TITLE
smoke tests: organize

### PR DESCRIPTION
**Pull request by midleman**
### Intent
Now that test parallelization is in place and working (🤞), I revisited the code to improve its organization and maintainability.

### Approach
* All helpers used to run the tests are now contained in `test/smoke/test-runner`.
* The actual script that kicks off the tests is now: `test/smoke/run-tests.js`

### QA Notes
* Confirmed full test suite and windows suite ran w/o issue.
* Confirmed logs are still there.
* Oh and turned retries back on.

### CODE REVIEW

File: test/smoke/tsconfig.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
 		"outDir": "out",
 		"sourceMap": true,
 		"skipLibCheck": true,
+		"declaration": true,
```
Comment (testlabauto) [inline test/smoke/tsconfig.json:14]: Probably should surround this with 
// --- Start Positron ---
// --- End Positron ---

**Review by testlabauto — DISMISSED** [review]
Looks good.  Just had one minor comment.

File: test/smoke/tsconfig.json (hunk 1 above)
Comment (midleman) [reply test/smoke/tsconfig.json:14]: Good catch, thank you!

**Review by testlabauto — APPROVED** [review]
LGTM!

**Pull request merged**